x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlAsyncGetResultsAction.java

```diff
@@ -40,6 +42,12 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         if (request.hasParam(KEEP_ALIVE_NAME)) {
             get.setKeepAlive(request.paramAsTime(KEEP_ALIVE_NAME, get.getKeepAlive()));
         }
-        return channel -> client.execute(SqlAsyncGetResultsAction.INSTANCE, get, new RestToXContentListener<>(channel));
+        return channel -> client.execute(SqlAsyncGetResultsAction.INSTANCE, get, new SqlResponseFormatter(channel, request));
     }
+
+    @Override
+    protected Set<String> responseParams() {
+        return Collections.singleton(URL_PARAM_DELIMITER);
```

Comment:
This seems strange, why do we return the Delimiter?

Reply:
That's used to signal that this param can be checked on, but not consumed. [BaseRestHandler.java#prepareRequest](https://github.com/elastic/elasticsearch/blob/c6c2f1bc3637433db7791414983f9cd1911399b0/server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java#L159) and responseParams() details how that works.